python/_tskitmodule.c

```diff
@@ -6073,6 +6073,86 @@ TableCollection_get_file_uuid(TableCollection *self, void *closure)
     return Py_BuildValue("s", self->tables->file_uuid);
 }
 
+static PyObject *
+TableCollection_get_metadata(TableCollection *self, void *closure)
+{
+    PyObject *ret = NULL;
+
+    /* Py_BuildValue returns Py_None for zero length, we would rather
+       return a zero-length string */
+    if (self->tables->metadata_length == 0) {
+        ret = PyBytes_FromString("");
+    } else {
+        ret = Py_BuildValue("y#", self->tables->metadata, self->tables->metadata_length);
+    }
+    return ret;
+}
+
+static int
+TableCollection_set_metadata(TableCollection *self, PyObject *arg, void *closure)
+{
+    int ret = -1;
+    int err;
+    char *metadata;
+    Py_ssize_t metadata_length;
+
+    if (arg == NULL) {
```

Comment:
Similarly, we're probably doing the same thing here for a bunch of tables/table collections. How about a function 
```c
if (parse_metadata_arg(arg, &metadata, &metadata_length) != 0){
    goto out;
}
At least then we're not rewriting the same error strings.
```

Reply:
I've factored this out for schemas, but as table collection metadata has a different type and message have left it as is.

Reply:
Nice, LGTM